app/components/blog/post/detail-client.tsx

```
@@ -66,7 +67,13 @@ export default function DetailClient({ post, recent }: DetailClientProps) {
   }
 
   // Extract sections from post content
-  const sections = post.body ? extractSections(post.body) : [];
+  // Extract sections with slug map reset
+  const sections = React.useMemo(() => {
```

Biome (2.1.2)

[error] This hook is being called conditionally, but all hooks must be called in the exact same order in every component render.

Hooks should not be called after an early return.

For React to preserve state between calls, hooks needs to be called unconditionally and always in the same order.
See https://reactjs.org/docs/hooks-rules.html#only-call-hooks-at-the-top-level

(lint/correctness/useHookAtTopLevel)